tests/test_data_genpdf_init.py

```diff
@@ -0,0 +1,180 @@
+            )
+        with lhapdf_path(tmp_path):
+            pdf = lhapdf.mkPDF("test_genpdf_parent_evolution_basis", 0)
+            for x in [0.1, 0.2, 0.5]:
+                for Q2 in [10.0, 20.0, 100.0]:
+                    np.testing.assert_allclose(
+                        pdf.xfxQ2(21, x, Q2), CT14.xfxQ2(21, x, Q2), rtol=3e-5
+                    )
+                    np.testing.assert_allclose(pdf.xfxQ2(2, x, Q2), 0.0)
+
+
+def test_genpdf_dict(tmp_path):
+    with cd(tmp_path):
+        genpdf.generate_pdf(
+            "test_genpdf_dict",
+            [21],
+            {
+                21: lambda x, Q2: 3.0 * x * (1.0 - x),
+                2: lambda x, Q2: 4.0 * x * (1.0 - x),
+            },
+        )
+        with lhapdf_path(tmp_path):
+            pdf = lhapdf.mkPDF("test_genpdf_dict", 0)
+            for x in [0.1, 0.2, 0.8]:
+                for Q2 in [10.0, 20.0, 100.0]:
+                    np.testing.assert_allclose(
+                        pdf.xfxQ2(21, x, Q2), 3.0 * x * (1.0 - x), rtol=3e-5
+                    )
+                    # 2 is available, but not active
+                    np.testing.assert_allclose(pdf.xfxQ2(2, x, Q2), 0.0)
+
+
+def test_genpdf_custom(tmp_path):
+    with cd(tmp_path):
+        c = np.zeros_like(br.flavor_basis_pids, dtype=np.float_)
+        c[br.flavor_basis_pids.index(1)] = 1.0
+        c[br.flavor_basis_pids.index(2)] = 0.5
+        genpdf.generate_pdf(
+            "test_genpdf_custom",
+            [c],
+            {
+                1: lambda x, Q2: 2.0 * x * (1.0 - x),
+                2: lambda x, Q2: -4.0 * x * (1.0 - x),
+            },
+            info_update={"ForcePositive": 0},
+        )
+        with lhapdf_path(tmp_path):
+            pdf = lhapdf.mkPDF("test_genpdf_custom", 0)
+            for x in [0.1, 0.2, 0.8]:
+                for Q2 in [10.0, 20.0, 100.0]:
+                    # np.testing.assert_allclose(
+                    #     pdf.xfxQ2(1, x, Q2),
+                    #     0.0,err_msg=f"pid=1,x={x},Q2={Q2}"
+                    # )
+                    np.testing.assert_allclose(
+                        pdf.xfxQ2(2, x, Q2), 0.0, err_msg=f"pid=2,x={x},Q2={Q2}"
+                    )
+
+
+def test_genpdf_MSTW_allflavors(tmp_path):
+    with cd(tmp_path):
+        with lhapdf_path(test_pdf):
+            # load reference PDFs
+            ref = {}
+            ref_head = {}
+            for mem in range(1 + 1):
+                ref[mem] = lhapdf.mkPDF("myMSTW2008nlo90cl", mem)
+                ref_head[mem] = genpdf.load.load_head_from_file(
+                    "myMSTW2008nlo90cl", mem
+                )
+            # filtering on all flavors
+            genpdf.generate_pdf(
+                "test_genpdf_MSTW_allflavors",
+                [21, 1, 2, 3, 4, 5, 6, -1, -2, -3, -4, -5, -6],
+                "myMSTW2008nlo90cl",
+                members=True,
+            )
+        with lhapdf_path(tmp_path):
+            for mem in range(1 + 1):
```

Comment:
Please check: PdfType got correctly inherited

Reply:
I think so. I also checked on the actual file in the tmp folder